src/relayclient/RelayClient.ts

```diff
@@ -136,12 +141,9 @@ export default class RelayClient {
     // TODO: should have a better strategy to decide how often to refresh known relays
     await this.knownRelaysManager.refresh()
     gsnTransactionDetails.gasPrice = gsnTransactionDetails.forceGasPrice ?? await this._calculateGasPrice()
-    if (gsnTransactionDetails.gasPrice == null) {
-      gsnTransactionDetails.gas = await this._calculateGasPrice()
-    }
     if (gsnTransactionDetails.gas == null) {
-      const estimated = await this.web3.eth.estimateGas(gsnTransactionDetails)
-      gsnTransactionDetails.gas = estimated.toString()
+      const estimated = await this.contractInteractor.estimateGas(gsnTransactionDetails)
+      gsnTransactionDetails.gas = `0x${estimated.toString(16)}`
```

Comment:
I'm sure there's `toHex()` somewhere, and if not, then need to create one

Reply:
We've got it for buffer and bignumber. This is just 'number'. We don't need number-to-hex usually, this is the only place I can think of it being used right now.
There is some legacy mess leftover with types of things. I fix these on the go.